### ISSUE TITLE
Examples table query problem with Postgresql

### ISSUE THREAD

**Issue by pongraczi** [laravel-enso/Enso#53]
Enso has an example "application", called examples and the route is: domain.tld/examples/table  
When I access to that example, the default query cause sql syntax errors with PGSQL.  

The generated query is:  
```sql
select count(*) as aggregate from "examples" where ("examples"."name" LIKE %% or "examples"."position" LIKE %% or "examples"."project" LIKE %% or "examples"."salary" LIKE %% or "examples"."taxes" LIKE %%)
```
#### Problems:  
* One general issue is the wildcard usage in the query: **%%**  
    It supposed to be like this: **'%%'**  

* The other problem is, the following operator type:  **INTEGER LIKE '%%'**  
    In other words, there is no such an operator, which compare an integer to a wildcard.  
    It happens with **"examples"."salary" LIKE '%%'** and **"examples"."taxes" LIKE '%%'**  
    It supposed to be like this: 
    **CAST("examples"."salary" AS TEXT) LIKE '%%' or CAST("examples"."taxes" AS TEXT) LIKE '%%'**

Is it a laravel issue or enso related?

**Comment by aocneanu** [comment]
Also a db grammar issue.

**Comment by pongraczi** [comment]
Is it pure Laravel problem? Should I report to Laravel framework?

**Comment by pongraczi** [comment]
Ok, I got it. Here is the code, where the filter set up:  
https://github.com/laravel-enso/VueDatatable/blob/414bfe810b93beddcb57d337cdf03b8e92344884/src/app/Classes/Table/Filters.php#L37   
Here if I change the line as shown below (+), the first problem disappears. Question is, does it compatible with MySql and others?
```php
 -   $query->orWhere($column->data, 'LIKE', '%'.$arg.'%');
 +   $query->orWhere($column->data, 'LIKE', '\'%'.$arg.'%\'');
```
Here one should check the column type and probably force a CAST or something else to avoid the 2nd case (INTEGER LIKE '%%').

**Comment by pongraczi** [comment]
When I get out the following from the json vuedatatable definition, the system is working. I have empty examples table, I will check it with some populated data.  
```
app/Http/Controllers/Examples# diff -u exampleTable.json-original exampleTable.json 
--- exampleTable.json-original	2018-03-01 23:08:25.304742497 +0100
+++ exampleTable.json	2018-03-01 23:08:50.612525722 +0100
@@ -75,7 +75,6 @@
             "name": "salary",
             "data": "examples.salary",
             "meta": [
-                "searchable",
                 "sortable",
                 "total"
             ]
@@ -84,7 +83,6 @@
             "name": "taxes",
             "data": "examples.taxes",
             "meta": [
-                "searchable",
                 "sortable",
                 "total"
             ]
@@ -106,4 +104,4 @@
             ]
         }
     ]
-}
\ No newline at end of file
+}
```

**Comment by pongraczi** [comment]
I guess, something still not ok.  
I populated the table with one row.  
The following sql query gives it back, as expected:  
```sql
select id as "dtRowId", name, position, seniority, project,
	            salary, taxes, is_active, hired_at from "examples" where ("examples"."name" LIKE '%%' or "examples"."position" LIKE '%%' or "examples"."project" LIKE '%%') limit 10 offset 0
```

But on the Examples table view, I got an empty result (without any sql syntax problem).  

![kijeloles_661](https://user-images.githubusercontent.com/5891983/36886133-18712c20-1deb-11e8-9d89-961308e1ad2a.png)

**Comment by aocneanu** [comment]
Sorry for the late response,

Indeed there was a problem with the search field where no entry was available. The builder was looking for `$request->has('search')` instead of `$request->filled('search')`

it's fixed in https://github.com/laravel-enso/VueDatatable/commit/0bc6d02e31bba6642f08e3bdb4e4e2b6466b4061

**Comment by pongraczi** [comment]
That's good, thank you!  (I did not check yet.)  
Any decision, regarding the **%%** vs **'%%'** ?

**Comment by pongraczi** [comment]
Test results:  
* Without any search term, now I got the list from the DB as expected.
* But using Search, I got the same issue, not valid syntax, explicit cast is necessary: 
    ```sql
    SQLSTATE[42883]: Undefined function: 7 ERROR: operator does not exist: integer ~~ unknown LINE 1: ...xamples"."project" LIKE $3 or "examples"."salary" LIKE $4 or... ^ HINT: No operator matches the given name and argument type(s). You might need to add explicit type casts. (SQL: select count(*) as aggregate from "examples" where ("examples"."name" LIKE '%john%' or "examples"."position" LIKE '%john%' or "examples"."project" LIKE '%john%' or "examples"."salary" LIKE '%john%' or "examples"."taxes" LIKE '%john%'))
```

**Comment by aocneanu** [comment]
@pongraczi 

`$query->orWhere($column->data, 'LIKE', '%'.$arg.'%');` is the line responsible for the query.

From what you are saying, if you try on postgre

```
User::orWhere('id', 'like', '%1%')->get()
```

you run in the error that you mention right?

(you can try the above in tinker)

**Comment by aocneanu** [comment]
I will close this for now, due to inactivity :)

**Comment by pongraczi** [comment]
Issue still exists in 2.3.8.
* Without cast, it is not possible to compare the string to Number with the LIKE operator, as explained before.

> The other problem is, the following operator type: INTEGER LIKE '%%'
> In other words, there is no such an operator, which compare an integer to a wildcard.
> It happens with "examples"."salary" LIKE '%%' and "examples"."taxes" LIKE '%%'
> It supposed to be like this:
> CAST("examples"."salary" AS TEXT) LIKE '%%' or CAST("examples"."taxes" AS TEXT) LIKE '%%'

I have no knowledge how to use tinker to test a theory, sorry.

**Comment by aocneanu** [comment]
@pongraczi you may be interested by #103 .

All the problems should be fixed starting with 2.7.7, I finally decided to rename all the affected columns.
